src/main/scala/com/horizon/exchangeapi/OrgsRoutes.scala

```diff
@@ -462,6 +469,55 @@ trait OrgsRoutes extends JacksonSupport with AuthenticationSupport {
     } // end of exchAuth
   }
 
+  /* ====== GET /orgs/{orgid}/search/nodes/error/all ================================ */
+  @GET
+  @Path("orgs/{orgid}/search/nodes/error/all")
+  @Operation(summary = "Returns all node errors", description = "Returns a list of all the node errors for an organization (that the caller has access to see) in an error state. Can be run by a user or agbot.",
+    parameters = Array(
+      new Parameter(name = "orgid", in = ParameterIn.PATH, description = "Organization id.")),
+    responses = Array(
+      new responses.ApiResponse(responseCode = "201", description = "response body:",
+        content = Array(new Content(schema = new Schema(implementation = classOf[AllNodeErrorsInOrgResp])))),
+      new responses.ApiResponse(responseCode = "400", description = "bad input"),
+      new responses.ApiResponse(responseCode = "401", description = "invalid credentials"),
+      new responses.ApiResponse(responseCode = "403", description = "access denied"),
+      new responses.ApiResponse(responseCode = "404", description = "not found")))
+  def nodeGetAllErrorsRoute: Route = (path("orgs" / Segment / "search"  / "nodes" / "error" / "all") & get) { orgid =>
+    logger.debug(s"Doing GET /orgs/$orgid/search/nodes/error/all")
+    exchAuth(TNode(OrgAndId(orgid,"#").toString),Access.READ) { ident =>
+      complete({
+        var queryParam = NodesTQ.rows.filter(_.orgid === orgid)
+        val userId = orgid + "/" + ident.getIdentity
+        ident match {
+          case _: IUser => if(!(ident.isSuperUser || ident.isAdmin)) queryParam = queryParam.filter(_.owner === userId)
+          case _ => ;
+        }
+        val q = for {
+          (ne, _) <- NodeErrorTQ.rows.filter(_.errors =!= "").filter(_.errors =!= "[]") join queryParam on (_.nodeId === _.id)
+        } yield (ne.nodeId, ne.errors, ne.lastUpdated)
+
+        db.run(q.result).map({ list =>
+          logger.debug("GET /orgs/"+orgid+"/search/nodes/error/all result size: "+list.size)
+          if (list.nonEmpty) {
+            val errorsMap = scala.collection.mutable.Map[String, NodeErrorsResp]()
+            for ((nodeId, errorsString, lastUpdated) <- list) {
+              errorsMap.get(nodeId) match {
```

Comment:
i'm unclear about why you have to do this processing (a comment would help): it seems like it is because the join can theoretically return multiple error rows for a nodeid. But the nodeerror table schema enforces that there is exactly 1 row per node, because of the nodeid foreign key. So it seems like this for loop could be done with a `.map()` method instead and avoid the ListBuffer. (Unless i'm missing something.)